src/SqlPersistence/Config/SqlPersistenceConfig_Connection.cs

```diff
@@ -25,13 +26,13 @@ public static void ConnectionBuilder(this PersistenceExtensions<SqlPersistence>
         /// Configures how <see cref="DbConnection"/>s are constructed, allowing for selecting a different database per tenant in a multi-tenant system.
         /// </summary>
         /// <param name="configuration"></param>
-        /// <param name="tenantIdHeaderName"></param>
-        /// <param name="buildConnectionFromTenantData"></param>
+        /// <param name="tenantIdHeaderName">The name of the message header that identifies the tenant id in each message.</param>
+        /// <param name="buildConnectionFromTenantData">Using a tenant id, builds a database connection for that tenant database.</param>
         public static void MultiTenantConnectionBuilder(this PersistenceExtensions<SqlPersistence> configuration, string tenantIdHeaderName, Func<string, DbConnection> buildConnectionFromTenantData)
```

Comment:
Is this "shortcut" needed now that we have the one with the `IncomingMessage`?

Reply:
I think it's the one that best leads to the pit of success, especially since dealing with the headers properly involves a TryGet. Even if you end up using the more complex one, having this overload tells you "I should be looking at headers for this, not trying to deserialize the message bytes or something."

One thing this brings up in my mind though is that we don't do anything if the multi-tenant builder gets a message with no tenant. It will either throw in the "identify header" stage or pass a null to the build connection stage. Perhaps we should throw a helpful exception? Because none of this is useful without a behavior to keep the tenant ids moving through message flows.

Reply:
I think an helpful exception is required.

> Because none of this is useful without a behavior to keep the tenant ids moving through message flows.

Is persistence responsibility to add such a behavior to the pipeline? I guess so. In which case the captured tenant id (and the headers) should be propagated to outgoing messages.

Reply:
I don't think it _can_ be the responsibility of the persistence.

If you use the `Func<IncomingMessage, string>` then how do you know which header to propagate into the next message? Your behavior would have to propagate any header that was touched and propagate them all.

To have that kind of info you'd need to remove the `Func<IncomingMessage, string>` option and have something more like `MultiTenantConnectionBuilder(string[] possibleHeadersContainingTenantId, Func<string, DbConnection> buildFromTenantId)`.

If you don't have that all the persistence can do is throw if it is expecting a tenant id and can't find one, prompting the developer to create (with guidance doco) their own behavior. We are in an advanced scenario here, after all.